class-wxr-import-ui.php

```diff
@@ -0,0 +1,635 @@
+
+		// Set uploader settings
+		wp_plupload_default_settings();
+		$settings = array(
+			'next_url' => wp_nonce_url( $this->get_url( 1 ), 'import-upload' ) . '&id={id}',
+			'plupload' => array(
+				'filter' => array(
+					'max_file_size' => $max_upload_size . 'b',
+				),
+
+				'file_data_name' => 'import',
+				'multipart_params' => array(
+					'action'   => 'wxr-import-upload',
+					'_wpnonce' => wp_create_nonce( 'wxr-import-upload' ),
+				),
+			),
+		);
+		wp_localize_script( 'import-upload', 'importUploadSettings', $settings );
+	}
+
+	/**
+	 * Display the author picker (or upload errors).
+	 */
+	protected function display_author_step() {
+		if ( isset( $_REQUEST['id'] ) ) {
+			$err = $this->handle_select( $_REQUEST['id'] );
+		} else {
+			$err = $this->handle_upload();
+		}
+		if ( is_wp_error( $err ) ) {
+			$this->display_error( $err );
+			return;
+		}
+
+		$data = $this->get_data_for_attachment( $this->id );
+		if ( is_wp_error( $data ) ) {
+			$this->display_error( $data );
+			return;
+		}
+
+		require __DIR__ . '/templates/select-options.php';
+	}
+
+	/**
+	 * Handles the WXR upload and initial parsing of the file to prepare for
+	 * displaying author import options
+	 *
+	 * @return bool|WP_Error True on success, error object otherwise.
+	 */
+	protected function handle_upload() {
+		$file = wp_import_handle_upload();
+
+		if ( isset( $file['error'] ) ) {
+			return new WP_Error( 'wxr_importer.upload.error', esc_html( $file['error'] ), $file );
+		} elseif ( ! file_exists( $file['file'] ) ) {
+			$message = sprintf(
+				__( 'The export file could not be found at <code>%s</code>. It is likely that this was caused by a permissions problem.', 'wordpress-importer' ),
+				esc_html( $file['file'] )
+			);
+			return new WP_Error( 'wxr_importer.upload.no_file', $message, $file );
+		}
+
+		$this->id = (int) $file['id'];
+		return true;
+	}
+
+	public function handle_async_upload() {
+		header( 'Content-Type: text/html; charset=' . get_option( 'blog_charset' ) );
+		send_nosniff_header();
+		nocache_headers();
+
+		check_ajax_referer( 'wxr-import-upload' );
+
+		/*
+		 * This function does not use wp_send_json_success() / wp_send_json_error()
+		 * as the html4 Plupload handler requires a text/html content-type for older IE.
+		 * See https://core.trac.wordpress.org/ticket/31037
+		 */
+
+		if ( ! current_user_can( 'upload_files' ) ) {
+			echo wp_json_encode( array(
+				'success' => false,
+				'data'    => array(
+					'message'  => __( 'You do not have permission to upload files.' ),
+					'filename' => $_FILES['import']['name'],
+				)
+			) );
+
+			exit;
+		}
+
+		$file = wp_import_handle_upload();
+		if ( is_wp_error( $file ) ) {
+			echo wp_json_encode( array(
+				'success' => false,
+				'data'    => array(
+					'message'  => $file->get_error_message(),
+					'filename' => $_FILES['import']['name'],
+				)
+			) );
+
+			wp_die();
+		}
+
+		if ( ! $attachment = wp_prepare_attachment_for_js( $file['id'] ) )
```

Comment:
Wasn't it something to do with this? https://blog.codecentric.de/en/2014/02/curly-braces/